src/main.c

```diff
@@ -443,1273 +451,1609 @@ typedef struct {
+                         uint64_t *result_size, const char **mime_type);
+void se_open_file_browser(bool clicked, float x, float y, float w, float h,
+                          void (*file_open_fn)(const char *dir),
+                          const char **file_types, char *output_path);
+void se_file_browser_accept(const char *path);
 void se_run_all_ar_cheats();
-void se_load_cheats(const char * filename);
-void se_save_cheats(const char* filename);
-void se_convert_cheat_code(char * text_code, int cheat_index);
+void se_load_cheats(const char *filename);
+void se_save_cheats(const char *filename);
+void se_convert_cheat_code(char *text_code, int cheat_index);
 static void se_reset_core();
-static bool se_load_theme_from_file(const char * filename);
-static bool se_draw_theme_region(int region, float x, float y, float w, float h);
-static bool se_draw_theme_region_tint(int region, float x, float y, float w, float h,uint32_t tint);
-static bool se_draw_theme_region_tint_partial(int region, float x, float y, float w, float h, float w_ratio, float h_ratio, uint32_t tint);
-static const char* se_get_pref_path(){
+static bool se_load_theme_from_file(const char *filename);
+static bool se_draw_theme_region(int region, float x, float y, float w,
+                                 float h);
+static bool se_draw_theme_region_tint(int region, float x, float y, float w,
+                                      float h, uint32_t tint);
+static bool se_draw_theme_region_tint_partial(int region, float x, float y,
+                                              float w, float h, float w_ratio,
+                                              float h_ratio, uint32_t tint);
+static const char *se_get_pref_path() {
 #if defined(EMSCRIPTEN)
   return "/offline/";
 #elif defined(USE_SDL)
-  static const char* cached_pref_path=NULL;
-  if(cached_pref_path==NULL)cached_pref_path=SDL_GetPrefPath("Sky","SkyEmu");
+  static const char *cached_pref_path = NULL;
+  if (cached_pref_path == NULL)
+    cached_pref_path = SDL_GetPrefPath("Sky", "SkyEmu");
   return cached_pref_path;
 #elif defined(PLATFORM_ANDROID)
-  ANativeActivity* activity =(ANativeActivity*)sapp_android_get_native_activity();
-  if(activity->internalDataPath)return activity->internalDataPath;
+  ANativeActivity *activity =
+      (ANativeActivity *)sapp_android_get_native_activity();
+  if (activity->internalDataPath)
+    return activity->internalDataPath;
 #endif
   return "";
 }
 #ifdef PLATFORM_ANDROID
 float se_android_get_display_dpi_scale();
 #endif
-static float se_dpi_scale(){
-  if(gui_state.dpi_override)return gui_state.dpi_override/120.;
+static float se_dpi_scale() {
+  if (gui_state.dpi_override)
+    return gui_state.dpi_override / 120.;
   static float dpi_scale = -1.0;
-  if(dpi_scale>0.)return dpi_scale;
+  if (dpi_scale > 0.)
+    return dpi_scale;
   dpi_scale = sapp_dpi_scale();
-  if(dpi_scale<=0)dpi_scale=1.;
-  dpi_scale*=1.10;
+  if (dpi_scale <= 0)
```

Comment:
1) if/while should not have spaces between "("
2) if statements without {} should always be on the same line as their statement (it prevents bugs)

Reply:
do we always want {} to create a new line?

Reply:
No, leave it up to the programmer to put the new lines if they want it.